### PR TITLE
Create a DB Class to mimic a Database

### PR DESCRIPTION
**Pull request by Denisqo**
Issue #41 has been resolved.

### CODE REVIEW

File: src/db/index.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        this.key = key
+        this.schema = schema
+    }
+
+    insert(entity) {
+        if (typeof entity !== 'object' || Object.keys(entity).length === 0 || Object.keys(entity).length !== this.schema.length) {
+            return new Error('Bad parameter "entity"')
+        }
+
+        if (!this.isSameKeys()) {
```
Comment (irshadjsr21) [inline src/db/index.js]: The function `isSameKeys` takes a parameter `entity` which is not passed here.

File: src/db/index.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
+class DB {
+    key;
+    schema;
+
+    constructor(key, schema) {
+        if (typeof key !== 'string' || typeof schema !== 'object' || key.length === 0 || schema.length === 0) {
+            return new Error('Bad parameter "key" or "schema"')
```
Comment (irshadjsr21) [inline src/db/index.js]: `throw` the Error instead of returning it. 

`return new Error()` => `throw new Error()`

File: src/db/index.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
+class DB {
+    key;
+    schema;
+
+    constructor(key, schema) {
+        if (typeof key !== 'string' || typeof schema !== 'object' || key.length === 0 || schema.length === 0) {
```
Comment (irshadjsr21) [inline src/db/index.js]: Please check here if the `schema` is an `Array` or not. In this check, even an object (ex: `{}`) will pass as a schema. 
To check for array you can use `Array.isArray(schema)`.

**Review by irshadjsr21 — CHANGES_REQUESTED** [review]

File: src/db/index.js (hunk 1 above)
Comment (Denisqo) [reply src/db/index.js]: Thanks!

**Comment by Denisqo** [comment]
Hello @irshadjsr21,

I solved the problems. Can we merge it?

**Review by irshadjsr21 — APPROVED** [review]

**Comment by irshadjsr21** [comment]
Greetings from Build With Girlscript Bangalore😃

Dear participant,

We have gone through your pull request made for issue #41  

Congratulations🎊 🎉 on getting your pull request successfully merged into our application Build with GSBlr-Raahat.

We will soon be distributing your valuable badge(s)🥇🥈🥉🎖️ corresponding to the pull requests that are merged. Please fill the participation form 📃 to receive your badges, if not already done.

Learning is a never-ending process. 📈 We will be coming up with new issues, stay tuned. Meanwhile, you can also check out our other Code with GirlScript Bangalore Repository to try some problem solving and earn exciting badges.

Thanks for your contribution! Happy coding😃

**Comment by girlscript-blr** [comment]
Hi @Denisqo 

Please fill the [participation form](https://tinyurl.com/buildwithgsblr) 📃  or share your email Id to receive your badges.

Thanks

**Pull request merged**